src/index.js

```diff
@@ -276,12 +278,14 @@ export default class Carousel extends Component {
    *
    * @param {Number} index - The slide index to move to.
    * @param {String} direction - The direction to transition, should be 'right' or 'left'.
+   * @param {Boolean} autoSlide - The source of slide transition, should be true for autoPlay and false for user click.
    */
-  goToSlide(index, direction) {
+  goToSlide(index, direction, autoSlide = false) {
     const { beforeChange, transitionDuration, transition, onControlClick } = this.props;
 
-    if (onControlClick) {
-      onControlClick();
+    if (!autoSlide && onControlClick) {
```

Comment:
ah hmm. I see. I wonder if `onControlClick` is the appropriate event here, since you want to fire it for non-"click" events as well.  really what you want is something like `onSlideTransitioned` . maybe something like:

```js
onSlideTransitioned({
  autoPlay: autoSlide,
  index,
  direction
});
```

That might capture enough data for consumers and remove some of the weirdness about autoplay here and naming, etc.